tests/Unit/Core/Form/IdentifiableObject/CommandBuilder/Product/ShippingCommandBuilderTest.php

```diff
@@ -31,27 +31,90 @@
 use PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\Product\ShippingCommandsBuilder;
 
 class ShippingCommandBuilderTest extends AbstractProductCommandBuilderTest
 {
     /**
-     * @dataProvider getExpectedCommands
+     * @dataProvider getExpectedSingleShopCommands
      *
      * @param array $formData
      * @param array $expectedCommands
      */
-    public function testBuildCommand(array $formData, array $expectedCommands): void
+    public function testBuildSingleShopCommands(array $formData, array $expectedCommands): void
     {
-        $builder = new ShippingCommandsBuilder();
-        $builtCommands = $builder->buildCommands($this->getProductId(), $formData);
+        $builder = new ShippingCommandsBuilder(self::MODIFY_ALL_SHOPS_PREFIX);
+        $builtCommands = $builder->buildCommands($this->getProductId(), $formData, $this->getSingleShopConstraint());
         $this->assertEquals($expectedCommands, $builtCommands);
     }
 
+    /**
+     * @dataProvider getExpectedAllShopCommands
+     *
+     * @param array $formData
+     * @param array $expectedCommands
+     */
+    public function testBuildAllShopCommands(array $formData, array $expectedCommands): void
+    {
+        $builder = new ShippingCommandsBuilder(self::MODIFY_ALL_SHOPS_PREFIX);
+        $builtCommands = $builder->buildCommands($this->getProductId(), $formData, $this->getSingleShopConstraint());
+        $this->assertEquals($expectedCommands, $builtCommands);
+    }
+
+    /**
+     * @return Generator
+     */
+    public function getExpectedAllShopCommands(): Generator
+    {
+        $command = $this->getAllShopsCommand();
+        /* @phpstan-ignore-next-line */
+        $command->setCarrierReferences([1, 2, 3]);
+        yield [
+            [
+                'shipping' => [
+                    'carriers' => ['1', '2', '3'],
+                    self::MODIFY_ALL_SHOPS_PREFIX . 'carriers' => true,
+                ],
+            ],
+            [$command],
+        ];
+
+        $command = $this->getAllShopsCommand();
+        $localizedNotes = [
+            '1' => 'test4',
+            '3' => 'test5',
+        ];
+        $command->setLocalizedDeliveryTimeOutOfStockNotes($localizedNotes);
+        yield [
+            [
+                'shipping' => [
+                    'delivery_time_notes' => [
+                        'out_of_stock' => $localizedNotes,
+                        self::MODIFY_ALL_SHOPS_PREFIX . 'out_of_stock' => true,
+                    ],
+                ],
+            ],
+            [$command],
+        ];
+
+        $command = $this->getAllShopsCommand();
+        $command->setAdditionalShippingCost('-0.55');
+        yield [
+            [
+                'shipping' => [
+                    'additional_shipping_cost' => '-0.55',
+                    self::MODIFY_ALL_SHOPS_PREFIX . 'additional_shipping_cost' => true,
+                ],
+            ],
+            [$command],
+        ];
```

Comment:
You should add one or two use cases when both commands are generated (single and all shops)